internal/dataplane/parser/translate_secrets.go

```diff
@@ -0,0 +1,102 @@
+package parser
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"time"
+
+	"github.com/kong/go-kong/kong"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/kongstate"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/store"
+)
+
+func getCACerts(log logrus.FieldLogger, storer store.Storer, plugins []kongstate.Plugin) []kong.CACertificate {
+	caCertSecrets, err := storer.ListCACerts()
+	if err != nil {
+		log.WithError(err).Error("failed to list CA certs")
+		return nil
+	}
+
+	var caCerts []kong.CACertificate
+	for _, certSecret := range caCertSecrets {
+		idBytes, ok := certSecret.Data["id"]
+		if !ok {
+			log.Errorf("skipping synchronisation, invalid CA certificate: missing 'id' field in data")
```

Comment:
Can we (attempt to) log something about the certificate so that the user can track down which certificate is the problematic one?

Reply:
Good point, this log wouldn't tell much: [83ce558](https://github.com/Kong/kubernetes-ingress-controller/pull/3063/commits/83ce558a74cc6b0b30957054fecf84e99a3ad48e)